WalletWasabi/Services/WalletService.cs

```diff
@@ -1245,9 +1257,19 @@ public Operation(Script script, Money amount, string label)
 				}
 			}
 
+			PSBT psbt = PSBT.FromTransaction(tx);
+			uint masterFP = BitConverter.ToUInt32(KeyManager.ExtPubKey.Fingerprint);
+			foreach (var coin in spentCoins.Concat(innerWalletOutputs).Concat(outerWalletOutputs).ToHashSet())
+			{
+				if (coin.HdPubKey != null)
+				{
+					psbt.AddKeyPath(coin.HdPubKey.PubKey, Tuple.Create(masterFP, coin.HdPubKey.NonHardenedKeyPath));
+				}
+			}
+
```

Comment:
@lontivero @NicolasDorier @joemphilips I'd like to create a PSBT here from my transaction. However coldcard tells me this:

"We require subpaths to be specified in the PSBT."

Even though I specified the subpaths: `coin.HdPubKey.NonHardenedKeyPath`

Reply:
I seemed to fix it by using `AddPathTo` instead of `AddKeyPath`, which I don't know why makes sense, anyway, going at the next error now.

Reply:
`AddPathTo` can add path information for arbitrary PubKey, while `AddKeyPath` is more safer that it can only add it for a related key.
Either is fine but usually `AddKeyPath` is better (since it is more safe). You will need `AddPathTo` when you want to add  key before adding [previous output | script] since logic for determining "related pubkey" depends on what has been added to the PSBTInput so far.
(I know the name is not great, there should be something better :( )

In your case, I think you were running AddKeyPath *before* adding previous output. so I suppose changing the order of methods will also fix the issue.